app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_results = {}
+
+    def wrapper(*args: (int, float, bool, str, bytes, tuple)) -> Any:
+        key = f"{func.__name__}: {args}"
```

Comment:
You could keep the whole function as an object to prevent the potential name collision:
```suggestion
        key = (func, args)
```